unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class_with_generic_inner_classes.cpp

```diff
@@ -0,0 +1,918 @@
+      }
+    }
+
+    THEN(
+      "The field3 component should be a pointer to "
+      "GenericClassWithGenericInnerClasses$GenericInnerClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field3");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(class_prefix + "$GenericInnerClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+      }
+    }
+
+    THEN(
+      "The field4 component should be a pointer to GenericClassWithGeneric"
+      "InnerClasses$GenericInnerClass$DoublyNestedInnerClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field4");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(
+          class_prefix + "$GenericInnerClass$DoublyNestedInnerClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Inst,
+            "java::Interface_Implementation"}});
+      }
+    }
+
+    THEN(
+      "The field5 component should be a pointer to GenericClassWithGeneric"
+      "InnerClasses$GenericInnerClass$DoublyNestedInnerClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field5");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(
+          class_prefix + "$GenericInnerClass$DoublyNestedInnerClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+      }
+    }
+
+    THEN(
+      "The field6 component should be a pointer to GenericClassWithGeneric"
+      "InnerClasses$GenericInnerClass$DoublyNestedInnerGenericClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field6");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(
+          class_prefix + "$GenericInnerClass$DoublyNestedInnerGenericClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
```

Comment:
Very minor nit: I'd be tempted to break this line after the `::Var,`, so that it matches the other components in this initializer list. It makes it easier to read at a glance of the layout style of all the elements is matching :-)